irods_specific_implementation/irods_rest_api_base.h

```diff
@@ -262,6 +329,8 @@ namespace irods::rest
         void configure_logger(const configuration& _cfg) noexcept
         {
             try {
+                logger_->set_pattern("[%Y-%m-%d %T.%e] [%n] [PID %P] [Thread %t] [%l] %v");
```

Comment:
@trel Here is the new format for log messages.

See spdlog's [documentation](https://github.com/gabime/spdlog/wiki/3.-Custom-formatting) for details.

Reply:
please add literal 'T'... rather than space... as per https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations

`[%Y-%m-%dT%T.%e]`

only other consideration would be to create structured logging / JSON rather than a single text line...  to be jq-friendly.

Reply:
and a possible trailing `Z` if we're not conveying timezone information (which... I think, we should not).

`[%Y-%m-%dT%T.%eZ]`

Reply:
I think we can adjust the formatting to be JSON. I'll give that a try.

Perhaps:
```javascript
{
    "log_level": "%l",
    "timestamp": "%Y-%m-%d",
    "logger": "%n",
    "pid": "%P",
    "thread": "%t",
    "message": "%v"
}
```

Reply:
Here is a message that was generated via the new format:
```javascript
{
  "timestamp": "2021-09-16T10:30:31.126-04:00",
  "logger": "irods_rest_cpp_zone_report_server",
  "pid": 14044,
  "thread": 14044,
  "severity": "trace",
  "message": "Endpoint initialized."
}
```
The pid/thread number are the same on startup. They change as requests are received.

Reply:
i like it all.   'severity'... timezone... only thing to think on is the word 'logger'...

Reply:
Alternatives for `"logger"`:
- binary
- service
- end_point
- source

Are there any other names?

Reply:
- api_endpoint
- service_endpoint
...

i think i like just 'service' ?

Reply:
`service` it is!

Reply:
done